release/rancher/rancher.go

```diff
@@ -296,3 +328,131 @@ func createPRFromRancher(ctx context.Context, rancherBaseBranch, title, branchNa
 
 	return err
 }
+
+type ContentLine struct {
+	Line    int
+	File    string
+	Content string
+}
+
+type Content struct {
+	RancherImages  []ContentLine
+	FilesWithRC    []ContentLine
+	MinFilesWithRC []ContentLine
+	ChartsWithDev  []ContentLine
+	KDMWithDev     []ContentLine
+}
+
+func CheckRancherRCDeps(ctx context.Context, local, forCi bool, org, repo, commitHash, files string) error {
+	var (
+		content  Content
+		badFiles bool
+	)
+
+	devDependencyPattern := regexp.MustCompile(`dev-v[0-9]+\.[0-9]+`)
+	rcTagPattern := regexp.MustCompile(`-rc[0-9]+`)
+
+	ghClient := repository.NewGithub(ctx, "")
+
+	for _, filePath := range strings.Split(files, ",") {
+		var scanner *bufio.Scanner
+		if local {
+			content, err := contentLocal(filePath)
+			if err != nil {
+				if os.IsNotExist(err) {
+					logrus.Debugf("file '%s' not found, skipping...", filePath)
+					continue
+				}
+				return err
+			}
+			defer content.Close()
+			scanner = bufio.NewScanner(content)
+		} else {
+			content, err := contentRemote(ctx, ghClient, org, repo, commitHash, filePath)
+			if err != nil {
+				return err
+			}
+			scanner = bufio.NewScanner(strings.NewReader(content))
+		}
+
+		lineNum := 1
+
+		for scanner.Scan() {
+			line := scanner.Text()
+			if devDependencyPattern.MatchString(line) {
+				lineContent := ContentLine{File: filePath, Line: lineNum, Content: formatContentLine(line)}
+				lineContentLower := strings.ToLower(lineContent.Content)
+				if strings.Contains(lineContentLower, "chart") {
+					badFiles = true
+					content.ChartsWithDev = append(content.ChartsWithDev, lineContent)
+				}
+				if strings.Contains(lineContentLower, "kdm") {
+					badFiles = true
+					content.KDMWithDev = append(content.KDMWithDev, lineContent)
+				}
+			}
+			if strings.Contains(filePath, "/package/Dockerfile") {
+				if !strings.Contains(line, "_VERSION") {
+					continue
+				}
```

Comment:
This and another condition are causing at least two bugs identifying dev kdm branches.
https://github.com/rancher/rancher/blob/release/v2.8/package/Dockerfile#L24
https://github.com/rancher/rancher/blob/release/v2.8/pkg/settings/setting.go#L84